### PR TITLE
pkgs/development: rename name to pname&version part 1

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
#103997

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/enumerate/default.nix [hunk 1]
```diff
@@ -6,11 +6,12 @@ if lib.versionAtLeast ocaml.version "4.06"
...
 
-stdenv.mkDerivation {
-  name = "ocaml-enumerate-111.08.00";
+stdenv.mkDerivation rec {
+  pname = "ocaml-enumerate";
+  version = "111.08.00";
 
   src = fetchurl {
-    url = "https://ocaml.janestreet.com/ocaml-core/111.08.00/individual/enumerate-111.08.00.tar.gz";
+    url = "https://ocaml.janestreet.com/ocaml-core/${lib.versions.majorMinor version}.00/individual/enumerate-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/ocaml-modules/enumerate/default.nix:14]:
```suggestion
    url = "https://ocaml.janestreet.com/ocaml-core/${version}/individual/enumerate-${version}.tar.gz";
```

**Review by SuperSandro2000 — APPROVED** [review]

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
autoconf 2.69 is the default autoconf used in nixpkgs.

**Comment by Stunkymonkey** [comment]
so this causes the massive rebuilds. should i separate it to another PR?

**Comment by SuperSandro2000** [comment]
> so this causes the massive rebuilds. should i separate it to another PR?

Yes, please try that so that we can put the other changes to master.

**Comment by SuperSandro2000** [comment]
Lets just target staging instead of fiddling around to know which package causes the mass rebuilds.

**Pull request merged**